Mathlib/RingTheory/Ideal/LocalRing.lean

```diff
@@ -344,6 +344,11 @@ theorem surjective_units_map_of_local_ringHom [CommRing R] [CommRing S] (f : R 
   exact hb
 #align local_ring.surjective_units_map_of_local_ring_hom LocalRing.surjective_units_map_of_local_ringHom
 
+/-- Every ring hom `f : K →+* R` from a field `K` to a nontrivial ring `R` is a local ring hom. -/
+instance (priority := 100) {K R} [Field K] [CommRing R] [Nontrivial R]
```

Comment:
Can you explain the need for the choice of priority?

Reply:
I think this is the case that the note [lower instance priority] will apply. This is an instance that will always apply during type-class resolution, so it should have a lower instance. I'll add the comments "-- see Note [lower instance priority]".